core/src/main/java/io/grpc/DnsNameResolver.java

```diff
@@ -161,7 +163,8 @@ public void run() {
             servers.add(
                 new ResolvedServerInfo(new InetSocketAddress(inetAddr, port), Attributes.EMPTY));
           }
-          savedListener.onUpdate(servers, Attributes.EMPTY);
+          savedListener.onUpdate(
+              Collections.<List<ResolvedServerInfo>>singletonList(servers),Attributes.EMPTY);
```

Comment:
Surprising about the checkstyle since it catches *everything* else :smiley: 

As for the generics, it seems like it shouldn't be needed, but without it I get

`error: incompatible types: List<ArrayList<ResolvedServerInfo>> cannot be converted to List<List<ResolvedServerInfo>>`